sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -70,7 +70,7 @@ def test_doit_square_MatrixSymbol_symsize():
 
 def test_doit_with_MatrixBase():
     X = ImmutableMatrix([[1, 2], [3, 4]])
-    assert MatPow(X, 0).doit() == ImmutableMatrix(Identity(2))
+    assert MatPow(X, 0).doit() == Identity(2)
```

Comment:
This test shouldn't change. If you take an instance of ```ImmutableMatrix``` to the zero power, you expect to get back an ```ImmutableMatrix```.